tensorflow/python/eager/context.py

```diff
@@ -1244,9 +1269,27 @@ def _initialize_physical_devices(self):
         return
 
       devs = pywrap_tfe.TF_ListPhysicalDevices()
-      self._physical_devices = [
-          PhysicalDevice(name=d.decode(),
-                         device_type=d.decode().split(":")[1]) for d in devs]
+      self._physical_devices = self._parse_physical_devices(devs)
+      self._physical_device_to_index = {
+          p: i for i, p in enumerate(self._physical_devices)
+      }
+
+      self._visible_device_list = list(self._physical_devices)
+      self._memory_growth_map = {
+          d: None for d in self._physical_devices if d.device_type == "GPU"
+      }
+
+    # Import device settings that may have been passed into the constructor
+    self._import_config()
+
+  def reinitialize_physical_devices(self):
```

Comment:
Implementation here seems almost the same as `_initialize_physical_devices`. Let's instead pass `reinitialize` parameter to `_initialize_physical_devices`. If it is `True`, then we will re-initialize even if `_physical_devices is not None`; if `False` we just make sure devices aren't initialized yet. By default, the parameter can be set to `False`.

Reply:
done